server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderUnitTests.java

```diff
@@ -130,6 +130,74 @@ public void testCanAllocateUsesMaxAvailableSpace() {
         );
     }
 
+    public void testAverageDiskUsage() {
+        var metadata = Metadata.builder()
+            .put(IndexMetadata.builder("index").settings(settings(IndexVersion.current())).numberOfShards(1).numberOfReplicas(0))
+            .build();
+        var clusterState = ClusterState.builder(ClusterName.DEFAULT)
+            .metadata(metadata)
+            .nodes(
+                DiscoveryNodes.builder()
+                    .add(DiscoveryNodeUtils.builder("node_0").roles(new HashSet<>(DiscoveryNodeRole.roles())).build())
+                    .add(DiscoveryNodeUtils.builder("node_1").roles(new HashSet<>(DiscoveryNodeRole.roles())).build())
+                    .add(DiscoveryNodeUtils.builder("node_2").roles(new HashSet<>(DiscoveryNodeRole.roles())).build())
```

Comment:
NIT:
```suggestion
                    .add(DiscoveryNodeUtils.create("node_0"))
                    .add(DiscoveryNodeUtils.create("node_1"))
                    .add(DiscoveryNodeUtils.create("node_2"))
```
I believe all roles are added by default so no need to manually set them